feat: resx docs and blogpost (#806)
tolgee/documentation: platform/formats/dotnet_resx.mdx

making them a standard choice for managing translations in .NET projects.
Learn more about .NET localization in the [official Microsoft documentation](https://learn.microsoft.com/en-us/dotnet/core/extensions/localization).

## Example
```xml
<root>
  <data name="welcome_message" xml:space="preserve">
    <value>Welcome to our app!</value>
  </data>
  <data name="user_count_message" xml:space="preserve">
    <value>There are {0} users online.</value>
  </data>
  <data name="weekdays" xml:space="preserve">
    <value>Sunday, Monday, Tuesday, Wednesday, Thursday, Friday, Saturday</value>
  </data>
</root>
```

## File structure
A .NET RESX file is an XML file with the following structure:

- **root**: The main container element for all translations.
- **data**: Each `data` element represents a translation key-value pair.
  - **name**: The unique identifier for the string.
  - **value**: The actual translation string.

## Placeholders
RESX files support placeholders for dynamic content. These placeholders are indexed using curly braces, e.g., `{0}`, `{1}`.
Since RESX placeholders are a subset of ICU, Tolgee imports RESX files assuming full ICU support.

> We recommend using a [library](https://www.nuget.org/packages/icu.net) that enables ICU placeholders in your .NET app to fully leverage this feature.

Reviewer: ```suggestion
For more complex message formatting, we recommend using a [library](https://www.nuget.org/packages/icu.net) that enables ICU placeholders in your .NET app to fully leverage ICU formatting features.
```

Author: ```suggestion
For more complex message formatting, we recommend using a [library](https://www.nuget.org/packages/icu.net) to fully leverage ICU formatting features in your .NET app.
```
To avoid duplicated "enables ICU" and "leverage ICU"
